adding a testing section and Python's Replayer (#1488)
temporalio/documentation: docs/app-dev-context/replays.md

---
id: replay
title: How to Replay a Workflow Execution

> description: Replay recreates the exact state of the Workflow code.

Reviewer: ```suggestion
description: Replay recreates the exact state of a Workflow Execution.
```
I have a mini-Dominik sitting on my shoulder.

sidebar_label: Replay
tags:
  - guide-context
---

> Replay recreates the exact state of the Workflow code. You can replay Workflows from the beginning of their history when resumed.
> 
> Replays allow code to resume only if it is compatible from a deterministic point of view.

Reviewer: ```suggestion
Replay recreates the exact state of a Workflow Execution.
You can replay a Workflow from the beginning of its history when resumed.

Replays allows code to resume only if it is compatible from a deterministic point of view.
```

> Replays allows code to resume only if it is compatible from a deterministic point of view.

Reviewer: ```suggestion
Replay allows code to resume only if it is compatible from a deterministic point of view.
```
Oops. My bad.

> **To retrieve the Workflow History**
> You can retrieve the Workflow History from any of the following options and then pass the object to your SDK of choice.

Reviewer: ```suggestion
To retrieve the Workflow History, use any of the following options and then pass the object to your SDK of choice.
```
Alternatively, add a blank line after the bold psuedoheading.

- Proto History object
- [Temporal Web UI](/web-ui)
  1. Select a Workflow ID.
  2. Click **Download**.

> - [tctl](/tctl) commands, for example:

Reviewer: ```suggestion
- [tctl](/tctl) commands; for example:
```
